src/Microsoft.VisualStudio.Threading.Analyzers.Tests/VSTHRD010MainThreadUsageAnalyzerTests.cs

```diff
@@ -71,13 +55,21 @@ class Test {
     IVsSolution Method() { return null; }
 }
 ";
-            this.expect.Locations = new[] { new DiagnosticResultLocation("Test0.cs", 7, 23, 7, 34) };
-            this.VerifyCSharpDiagnostic(test, this.expect);
-            this.VerifyCSharpFix(test, fix);
+            var expected = Verify.Diagnostic(DescriptorSync).WithSpan(7, 23, 7, 34).WithArguments("IVsSolution", "Test.VerifyOnUIThread");
+            await new Verify.Test
+            {
+                TestCode = test,
+                ExpectedDiagnostics = { expected },
+                FixedCode = fix,
+
+                // Expected: SimpleMemberAccessExpression
+                // Actual:   QualifiedName
+                CodeFixValidationMode = CodeFixValidationMode.None,
+            }.RunAsync();
```

Comment:
Is this ignoring a bug in the analyzer, where we construct the wrong syntax tree? That's pretty cool if the new testing framework catches that. :) But shouldn't we fix it rather than ignore it here?

Reply:
https://github.com/Microsoft/vs-threading/issues/373